release/release-scripts/go-release.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+# Helper script to prepare a Go Release for the SDKs.
+
+set -e
+
+enforce_latest_code() {
+    if [[ -n "$(git status --porcelain=v1)" ]]; then
+        echo "ERROR: working directory is not clean."
+        echo "Please stash your changes and try again."
+        exit 1
+    fi
+
+    git fetch --quiet origin main
+    if [[ "$(git rev-parse HEAD)" != "$(git rev-parse origin/main)" ]]; then
+        echo "ERROR: This script was not run from the latest code from origin/main."
+        echo "Make sure to update your git branch with the latest from main and try again."
+        exit 1
+    fi
+}
```

Comment:
This is only useful in the local rc creation use-case, as I highlighted in my comment above - let's make sure we structure the code in such a way that this is clear, as well as expose the proper make targets for local scenarios.